Classes/Hooks/HookProvider.php

```diff
@@ -123,6 +124,26 @@ public function executeHookReturningMergedArray(string $method, ...$params): arr
         return \array_merge([], ...$result);
     }
 
+    /**
+     * Executes the hooked-in methods that pass on a manipulated value.
+     *
+     * @param string $method the method to execute
+     * @param mixed $value the value to manipulate by `$method()`
+     * @param mixed $params parameters to `$method()`
+     *
+     * @return mixed the manipulated value
+     */
+    public function executeHookReturningModifiedValue(string $method, $value, ...$params)
```

Comment:
Please don't modify or overwrite arguments in a method. Instead, please copy its value into a new local variable. (This makes reading the method easier as the reader usually assumes that `$value` is the parameter, not a new variable.)